### ISSUE TITLE
RNNs do not handle padding correctly

### ISSUE THREAD

**Issue by pplantinga** [speechbrain/speechbrain#501]
This is an issue discovered with the help of @JasonSWFu , the behavior of RNNs is different based on the batch size, presumably due to padding differences. I'm pretty sure the intended way to handle variable-length batches in PyTorch is to use `torch.nn.utils.rnn.pack_padded_sequence` on tensors before passing them to RNNs, so that padding is ignored.

The two approaches to solving it that came to mind:

1. Add `pack_padded_sequence` to each RNN's `forward()` method. Unfortunately this function requires the length, so this approach also requires updating every call to an RNN module, including those inside of `Sequential` containers.
2. Warn users when an RNN gets a tensor rather than a `PackedSequence` object, so they can manually add the `pad_packed_sequence` call. Unfortunately, this approach also doesn't handle `Sequential` containers.

This seems common enough that perhaps some sort of intervention in `Sequential` is appropriate, like always passing the length if there are two parameters to the forward method.

I think this issue is pretty important, how should we proceed with fixing it?
